icu4j/main/classes/core/src/com/ibm/icu/impl/number/UsagePrefsHandler.java

```diff
@@ -0,0 +1,121 @@
+// © 2020 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+package com.ibm.icu.impl.number;
+
+import com.ibm.icu.impl.IllegalIcuArgumentException;
+import com.ibm.icu.impl.units.MeasureUnitImpl;
+import com.ibm.icu.impl.units.UnitsRouter;
+import com.ibm.icu.number.Precision;
+import com.ibm.icu.util.Measure;
+import com.ibm.icu.util.MeasureUnit;
+import com.ibm.icu.util.ULocale;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UsagePrefsHandler implements MicroPropsGenerator {
+
+    private final MicroPropsGenerator fParent;
+    private UnitsRouter fUnitsRouter;
+
+    public UsagePrefsHandler(ULocale locale, MeasureUnit inputUnit, String usage, MicroPropsGenerator parent) {
+        assert parent != null;
+
+        this.fParent = parent;
+        this.fUnitsRouter =
+                new UnitsRouter(MeasureUnitImpl.forIdentifier(inputUnit.getIdentifier()), locale.getCountry(), usage);
+    }
+
+    private static Precision parseSkeletonToPrecision(String precisionSkeleton) {
+        final String kSuffixPrefix = "precision-increment/";
+        if (!precisionSkeleton.startsWith(kSuffixPrefix)) {
+            throw new IllegalIcuArgumentException("precisionSkeleton is only precision-increment");
+        }
+
+        String skeleton = precisionSkeleton.substring(kSuffixPrefix.length());
+        String skeletons[] = skeleton.split("/");
```

Comment:
Here's a function I failed to review! :)

I don't understand what this .split("/") is for. "precision-increment/NNN" is already reduced to "NNN" by the substring() function.

Reply:
Below, with "den", it looks like this code is expecting a "precision-increment/NNN/MMM" - which as far as I know is not supported in precision-increment skeletons.

Reply:
#1337 is fixing this, I noticed as I was moving it to UnitsRouter.